components/gitpod-db/src/typeorm/installation-admin-db-impl.ts

```diff
@@ -55,4 +49,14 @@ export class TypeORMInstallationAdminImpl implements InstallationAdminDB {
         /* Record not found - create one */
         return this.createDefaultRecord();
     }
+
+    async setSettings(settings: InstallationAdminSettings): Promise<void> {
+        const record = await this.getData();
+        record.settings = {
+            ...settings,
```

Comment:
@MrSimonEmms This `...` might be an attack vector, depending on the implementation of `repo.save(...)`. (we had one with `.update` last fall).

I suggest we:
 - keep the spread operator (`...`) here
 - add an explicit loop over all fields in `api.adminUpdateSettings` (cmp. comment below)